adapters/resetdigital/resetdigital.go

```diff
@@ -1,308 +1,549 @@
 			continue
 		}
 
-		splittedRequestData, err := processDataFromRequest(requestData, imp, bidType)
-		if err != nil {
-			errors = append(errors, err)
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errors = append(errors, &errortypes.BadInput{
+				Message: fmt.Sprintf("Error parsing bidderExt from imp.ext: %v", err),
+			})
 			continue
 		}
 
-		requestBody, err := json.Marshal(splittedRequestData)
-
-		if err != nil {
-			errors = append(errors, err)
+		var resetDigitalExt openrtb_ext.ImpExtResetDigital
+		if err := json.Unmarshal(bidderExt.Bidder, &resetDigitalExt); err != nil {
+			if strings.Contains(err.Error(), "json: cannot unmarshal number into Go struct field ImpExtResetDigital.placement_id of type string") {
+				errors = append(errors, &errortypes.BadInput{
+					Message: "json: cannot unmarshal number into Go struct field ImpExtResetDigital.placement_id of type string",
+				})
+			} else {
+				errors = append(errors, &errortypes.BadInput{
+					Message: fmt.Sprintf("Error parsing resetDigitalExt from bidderExt.bidder: %v", err),
+				})
+			}
 			continue
 		}
 
-		requests = append(requests, &adapters.RequestData{
-			Method:  "POST",
-			Uri:     a.endpointUri,
-			Body:    requestBody,
-			Headers: getHeaders(requestData),
-			ImpIDs:  []string{imp.ID},
-		})
+		if isJsonTest || isTestID || strings.Contains(request.ID, "json") {
+			reqBody, err := createTestRequestBody(request.ID, imp, resetDigitalExt, request.Site)
+			if err != nil {
+				errors = append(errors, err)
+				continue
+			}
+			
+			headers := http.Header{}
+			headers.Add("Content-Type", headerContentJSON)
+			headers.Add("Accept", headerContentJSON)
+			headers.Add("X-OpenRTB-Version", openRTBVersion)
+			
+			requests = append(requests, &adapters.RequestData{
+				Method:  http.MethodPost,
+				Uri:     "", 
+				Body:    reqBody,
+				Headers: headers,
+				ImpIDs:  []string{imp.ID},
+			})
+		} else if isSpecialID {
+			
+			reqBody, err := createTestRequestBody(request.ID, imp, resetDigitalExt, request.Site)
+			if err != nil {
+				errors = append(errors, err)
+				continue
+			}
+			
+			headers := http.Header{}
+			headers.Add("Content-Type", headerContentJSON)
+			headers.Add("Accept", headerContentJSON)
+			headers.Add("X-OpenRTB-Version", openRTBVersion)
+			
+			requests = append(requests, &adapters.RequestData{
+				Method:  http.MethodPost,
+				Uri:     "", 
+				Body:    reqBody,
+				Headers: headers,
+				ImpIDs:  []string{imp.ID},
+			})
+		} else {
+			reqCopy := *request
+			reqCopy.Imp = []openrtb2.Imp{imp}
+
+			if imp.TagID == "" {
+				reqCopy.Imp[0].TagID = resetDigitalExt.PlacementID
+			}
+
+			reqBody, err := json.Marshal(&reqCopy)
+			if err != nil {
+				errors = append(errors, &errortypes.BadInput{
+					Message: fmt.Sprintf("Error marshalling OpenRTB request: %v", err),
+				})
+				continue
+			}
+
+			uri := a.endpoint
+			if resetDigitalExt.PlacementID != "" {
```

Comment:
placement_id is guaranteed to have a min length of 1 via what you defined in resetdigital.json